{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "fc4545ea",
   "metadata": {},
   "source": [
    "## Plan\n",
    "1. Convert BAM files of HIFI reads to fastq (tool: bam2fastx)\n",
    "2. Evaluate quality of reads with Jellyfish and Genomescope => kmer historam as output\n",
    "3. Evaluete ploidy with Smurdgeplot => kmer-plot with ploidy\n",
    "4. Genome assembly with Flye, Hifiasm, LJA => results in the table ()\n",
    "5. Quality control of genome assembly with Quast and BUSCO => result also in the table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d23cdc84",
   "metadata": {},
   "outputs": [],
   "source": [
    "#1. Convert BAM files of HIFI reads to fastq (tool: bam2fastx)\n",
    "\n",
    "input_file_1 = '/media/eternus1/data/plants/boechera_falcata/raw_reads/hifi/BOFA_3455_P2_m64165_220403_044341.hifi.bam'\n",
    "output_file_1 = '/media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64165_220403_044341.hifi.fastq.gz'\n",
    "command = 'bam2fastq -o /media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64165_220403_044341.hifi /media/eternus1/data/plants/boechera_falcata/raw_reads/hifi/BOFA_3455_P2_m64165_220403_044341.hifi.bam'\n",
    "\n",
    "input_file_2 = '/media/eternus1/data/plants/boechera_falcata/raw_reads/hifi/BOFA_3455_P2_m64284e_220309_165827.hifi.bam'\n",
    "output_file_2 = '/media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64284e_220309_165827.hifi.fastq.gz'\n",
    "command = 'bam2fastq -o /media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64284e_220309_165827.hifi /media/eternus1/data/plants/boechera_falcata/raw_reads/hifi/BOFA_3455_P2_m64284e_220309_165827.hifi.bam'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c18871e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#2.1. FASTQC\n",
    "\n",
    "input_file_1 = '/media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64165_220403_044341.hifi.fastq.gz'\n",
    "output_file_1 = '/fastqc/BOFA_3455_P2_m64165_220403_044341.hifi_fastqc.html'\n",
    "command = 'fastqc -o fastqc /media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64165_220403_044341.hifi.fastq.gz'\n",
    "\n",
    "input_file_2 = '/media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64284e_220309_165827.hifi.fastq.gz'\n",
    "output_file_2 = '/fastqc/BOFA_3455_P2_m64284e_220309_165827.hifi_fastqc.html'\n",
    "command = 'fastqc -o fastqc_e /media/eternus1/data/plants/boechera_falcata/users/burdaeva/hifi_fastq/BOFA_3455_P2_m64284e_220309_165827.hifi.fastq.gz'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ccbd3d2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "BOFA_3455_P2_m64165_220403_044341.hifi.fastq.gz\n",
    "\n",
    "#2.1.1. Per base sequence quality\n",
    "\n",
    "Good.\n",
    "Calls have a good quality (they are in the green zone). \n",
    "There is no need for trimming (?)\n",
    "\n",
    "#2.1.2. Per sequence quality scores\n",
    "\n",
    "Good.\n",
    "The majority of reads have a high average quality score with no large bumps at the lower quality values.\n",
    "\n",
    "#2.1.3. Per base sequence content\n",
    "\n",
    "Good.\n",
    "The lines run parallel in general. \n",
    "\n",
    "#2.1.4. Per sequence GC content\n",
    "\n",
    "Failure.\n",
    "There is a high content of GC (why is there a big difference between the theory and reality?)\n",
    "There is also a sharp peak on a normal distribution (contamination?) \n",
    "\n",
    "#2.1.5. Per base N content\n",
    "\n",
    "Good.\n",
    "There is no N (a sequencer made all base calls with sufficient confidence)\n",
    "\n",
    "#2.1.6. Sequence Length Distribution\n",
    "\n",
    "Warning.\n",
    "All sequences are not the same length (?; it depends on the fragment size selection, primers, DNA weight and quality, etc.)\n",
    "\n",
    "#2.1.7. Sequence Duplication Levels\n",
    "                                       \n",
    "Good.\n",
    "Low level of duplication => high level of coverage of the target sequence (?)\n",
    "\n",
    "#2.1.8. Overrepresented sequences\n",
    "\n",
    "Good.\n",
    "No overrepresented sequences.\n",
    "\n",
    "#2.1.9. Adapter Content\n",
    "                 \n",
    "Good.\n",
    "No adapters (?)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6df4bfcf",
   "metadata": {},
   "outputs": [],
   "source": [
    "#3. Evaluate quality of reads with Jellyfish and Genomescope => kmer historam as output\n",
    "\n",
    "#3.1. Unzip\n",
    "\n",
    "input_file_1 ='BOFA_3455_P2_m64165_220403_044341.hifi.fastq.gz'\n",
    "output_file_1 ='hifi_1'\n",
    "command = 'gzip -dk BOFA_3455_P2_m64165_220403_044341.hifi.fastq.gz'\n",
    "\n",
    "input_file_2 ='BOFA_3455_P2_m64284e_220309_165827.hifi.fastq.gz'\n",
    "output_file_2 ='hifi_e'\n",
    "command = 'gzip -dk BOFA_3455_P2_m64284e_220309_165827.hifi.fastq.gz'\n",
    "\n",
    "#3.2. Merge files\n",
    "\n",
    "input_file_1 ='hifi_1'\n",
    "input_file_2 ='hifi_e'\n",
    "output_file ='hifi_1e'\n",
    "command = 'cat hifi_1 hifi_e > hifi_1e'\n",
    "\n",
    "#3.3. Jellyfish count\n",
    "\n",
    "input_file ='hifi_1e'\n",
    "output_file ='hifi_count'\n",
    "command = 'jellyfish count -m 21 -o ./hifi_count -s 23G -t 96 -C ./hifi_1e'\n",
    "\n",
    "#3.4. Jellyfish histo\n",
    "\n",
    "input_file ='hifi_count'\n",
    "output_file ='hifi.histo'\n",
    "command = 'jellyfish histo -o hifi.histo hifi_count'\n",
    "\n",
    "#3.5.1. Genomescope\n",
    "\n",
    "input_file ='hifi.histo'\n",
    "output_dir ='hifi_genomescope'\n",
    "command = 'genomescope2 -i hifi.histo -o hifi_genomescope -k 21'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6870c41",
   "metadata": {},
   "outputs": [],
   "source": [
    "#3.5.2. Plots genomescope\n",
    "\n",
    "Poisson profile -> homozygous genome\n",
    "Peak 130 is the coverage with the highest number of different 21-mers (average coverage depth). (это обозначается как 130х?)\n",
    "Additional peaks: \n",
    "low frequency false k-mers - sequencing errors (0.333 %)\n",
    "higher k-mer frequencies - repeats (0.328)\n",
    "Genome length: 90 360 679 bp\n",
    "Homozygosity: 86,8 %\n",
    "Heterozygosity: 13,2 %\n",
    "Ploidy: 2 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33aba892",
   "metadata": {},
   "outputs": [],
   "source": [
    "#4. Evaluete ploidy with Smudgeplot => kmer-plot with ploidy\n",
    "\n",
    "#4.1. Extraction of genomic kmers\n",
    "\n",
    "L=$(smudgeplot.py cutoff hifi.histo L)\n",
    "U=$(smudgeplot.py cutoff hifi.histo U)\n",
    "echo $L $U\n",
    "[60 2000]\n",
    "jellyfish dump -c -L $L -U $U ./hifi_count | smudgeplot.py hetkmers -o kmer_pairs\n",
    "output_files = 'kmer_pairs_coverages.tsv' and 'kmer_pairs_sequences.tsv'\n",
    "\n",
    "#4.2. Smudgeplot\n",
    "\n",
    "smudgeplot.py plot kmer_pairs_coverages.tsv -o smudgeplot\n",
    "output_dir = 'smudgeplot'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c4e70b52",
   "metadata": {},
   "outputs": [],
   "source": [
    "#4.3. Plots smudgeplot\n",
    "\n",
    "Ploidy: 2 # AB smudge is the brightest?)\n",
    "High homozygosity # peak 131?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06e3ad4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#5. Genome assembly => results in the table (https://docs.google.com/spreadsheets/d/1FeYK-oIFX5Gjc8w9Y93YdnkQcxDknQ_PTg7Af2crN4g/edit#gid=0)\n",
    "\n",
    "(hifi_1e = hifi_1e.fastq)\n",
    "\n",
    "#5.1. HiFiasm\n",
    "\n",
    "input_file = 'hifi_1e'\n",
    "output_files = 'hifi.asm'\n",
    "command = 'hifiasm -o hifi.asm -t 96 hifi_1e'\n",
    "\n",
    "#5.2. Canu\n",
    "\n",
    "input_file_90 = 'hifi_1e'\n",
    "output_files_90 = 'hifi_canu'\n",
    "command_90 = 'canu -p canu_hifi -d ./Canu genomeSize=90m -pacbio-raw ./main_file/hifi_1e useGrid=false'\n",
    "\n",
    "input_file_200 = 'hifi_1e.fastq'\n",
    "output_files_200 = 'Canu_200'\n",
    "command_200 = 'canu -p Canu_200 -d ./assembly_new/Canu_200 genomeSize=200m -pacbio-raw ./main_file/hifi_1e.fastq useGrid=false'\n",
    "\n",
    "#5.3. Peregrine\n",
    "\n",
    "input_file ='reads.lst' #file with the path to hifi_1e\n",
    "output_files = 'asm'\n",
    "command = 'pg_asm -k 21 reads.lst ./asm'\n",
    "\n",
    "#5.4. Wtdbg2\n",
    "\n",
    "input_file_1_90 = 'hifi_1e'\n",
    "output_files_1_90 = 'Wtdbg2'\n",
    "command_1_90 = 'wtdbg2 -x ccs -g 90m -t 96 -i ./main_file/hifi_1e -fo ./Wtdbg2/hifi_wtdbg'\n",
    "\n",
    "input_file_2_90 = 'hifi_wtdbg.ctg.lay.gz' #from Wtdbg2\n",
    "output_file_2_90 = 'hifi_wtpoa_cns.ctg.fa'\n",
    "command_2_90 = 'wtpoa-cns -t 96 -i ./hifi_wtdbg.ctg.lay.gz -fo ./cns/hifi_wtpoa_cns.ctg.fa'\n",
    "\n",
    "input_file_1_200 = 'hifi_1e.fastq'\n",
    "output_files_1_200 = 'Wtdbg2_200'\n",
    "command_1_200 = 'wtdbg2 -x ccs -g 200m -t 96 -i ./main_file/hifi_1e.fastq -fo ./assembly_new/Wtdbg2_200/hifi_wtdbg_200'\n",
    "\n",
    "input_file_2_200 = 'hifi_wtdbg_200.ctg.lay.gz' #from Wtdbg2_200\n",
    "output_file_2_200 = 'hifi_wtpoa_cns_200.ctg.fa'\n",
    "command_2_200 = 'wtpoa-cns -t 96 -i ./hifi_wtdbg_200.ctg.lay.gz -fo ./cns_200/hifi_wtpoa_cns_200.ctg.fa'\n",
    "\n",
    "#5.4. Falcon\n",
    "\n",
    "\n",
    "\n",
    "#5.5. LGA\n",
    "\n",
    "input_file = 'hifi_1e.fastq' \n",
    "output_file = 'LJA'\n",
    "command = './LJA/bin/lja --diploid -o ./assembly_new/LJA --reads ./main_file/hifi_1e.fastq'\n",
    "\n",
    "#5.6. Flye\n",
    "\n",
    "flye --pacbio-hifi ./main_file/hifi_1e.fastq -o ./assembly_new/Flye -g 200m -t 96 --scaffold      \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11f64bf4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#6. Quality control of genome assembly with Quast and BUSCO => result also in the table\n",
    "\n",
    "#6.1. QUAST\n",
    "\n",
    "input_file = 'canu_hifi.contigs.fasta'\n",
    "output_files = 'QUAST_Canu'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_Canu ./assembly_new/Canu/canu_hifi.contigs.fasta'\n",
    "\n",
    "input_file = 'hifi_wtpoa_cns.ctg.fa'\n",
    "output_files = 'QUAST_Wtdbg'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_Wtdbg ./assembly_new/Wtdbg2/cns/hifi_wtpoa_cns.ctg.fa'\n",
    "\n",
    "input_file = 'hifi.asm.bp.p_ctg.fa'\n",
    "output_files = 'QUAST_HiFiasm'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_HiFiasm ./assembly_new/HiFiasm/hifi.asm.bp.p_ctg.fa'\n",
    "\n",
    "input_file = 'asm_ctgs_m.fa'\n",
    "output_files = 'QUAST_Peregrine'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_Peregrine ./assembly_new/Peregrine/asm/asm_ctgs_m.fa'\n",
    "\n",
    "input_file = 'assembly.fasta'\n",
    "output_files = 'QUAST_LJA'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_LJA ./assembly_new/LJA/assembly.fasta'\n",
    "\n",
    "input_file = 'Canu_200.contigs.fasta'\n",
    "output_files = 'QUAST_Canu_200'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_Canu_200 ./assembly_new/Canu_200/Canu_200.contigs.fasta'\n",
    "\n",
    "input_file = 'hifi_wtpoa_cns_200.ctg.fa'\n",
    "output_files = 'QUAST_Wtdbg_200'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_Wtdbg_200 ./assembly_new/Wtdbg2_200/cns_200/hifi_wtpoa_cns_200.ctg.fa'\n",
    "\n",
    "input_file = 'assembly.fasta'\n",
    "output_files = 'QUAST_Flye'\n",
    "command = 'quast.py -o ./QC_after_as_raw/QUAST/QUAST_Flye ./assembly_new/Flye/assembly.fasta'\n",
    "\n",
    "#6.2. BUSCO\n",
    "\n",
    "#kapusta\n",
    "\n",
    "input_file = 'canu_hifi.contigs.fasta'\n",
    "output_files = 'BUSCO_Canu'\n",
    "command ='busco -m genome -i ./assembly_new/Canu/canu_hifi.contigs.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_Canu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "input_file = 'hifi_wtpoa_cns.ctg.fa'\n",
    "output_files = 'BUSCO_Wtdbg'\n",
    "command = 'busco -m genome -i ./assembly_new/Wtdbg2/cns/hifi_wtpoa_cns.ctg.fa -o ./QC_after_as_raw/BUSCO/BUSCO_Wtdbg -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "input_file = 'asm_ctgs_m.fa'\n",
    "output_files = 'BUSCO_Peregrine'\n",
    "command = 'busco -m genome -i ./assembly_new/Peregrine/asm/asm_ctgs_m.fa -o ./QC_after_as_raw/BUSCO/BUSCO_Peregrine -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "input_file = 'hifi.asm.bp.p_ctg.fa'\n",
    "output_files = 'BUSCO_HiFiasm'\n",
    "command = 'busco -m genome -i ./assembly_new/HiFiasm/hifi.asm.bp.p_ctg.fa -o ./QC_after_as_raw/BUSCO/BUSCO_HiFiasm -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "input_file = 'assembly.fasta'\n",
    "output_files = 'BUSCO_LJA'\n",
    "command = 'busco -m genome -i ./assembly_new/LJA/assembly.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_LJA -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "input_file = 'Canu_200.contigs.fasta'\n",
    "output_files = 'BUSCO_Canu_200'\n",
    "command ='busco -m genome -i ./assembly_new/Canu_200/Canu_200.contigs.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_Canu_200 -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "input_file = 'hifi_wtpoa_cns_200.ctg.fa'\n",
    "output_files = 'BUSCO_Wtdbg_200'\n",
    "command = 'busco -m genome -i ./assembly_new/Wtdbg2_200/cns_200/hifi_wtpoa_cns_200.ctg.fa -o ./QC_after_as_raw/BUSCO/BUSCO_Wtdbg_200 -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "input_file = 'assembly.fasta'\n",
    "output_files = 'BUSCO_Flye'\n",
    "command = 'busco -m genome -i ./assembly_new/Flye/assembly.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_Flye -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/brassicales_odb10'\n",
    "\n",
    "#eukaryotes\n",
    "\n",
    "input_file = 'canu_hifi.contigs.fasta'\n",
    "output_files = 'BUSCO_Canu_eu'\n",
    "command ='busco -m genome -i ./assembly_new/Canu/canu_hifi.contigs.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_Canu_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'\n",
    "\n",
    "input_file = 'hifi_wtpoa_cns.ctg.fa'\n",
    "output_files = 'BUSCO_Wtdbg_eu'\n",
    "command = 'busco -m genome -i ./assembly_new/Wtdbg2/cns/hifi_wtpoa_cns.ctg.fa -o ./QC_after_as_raw/BUSCO/BUSCO_Wtdbg_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'\n",
    "\n",
    "input_file = 'asm_ctgs_m.fa'\n",
    "output_files = 'BUSCO_Peregrine_eu'\n",
    "command = 'busco -m genome -i ./assembly_new/Peregrine/asm/asm_ctgs_m.fa -o ./QC_after_as_raw/BUSCO/BUSCO_Peregrine_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'\n",
    "\n",
    "input_file = 'hifi.asm.bp.p_ctg.fa'\n",
    "output_files = 'BUSCO_HiFiasm_eu'\n",
    "command = 'busco -m genome -i ./assembly_new/HiFiasm/hifi.asm.bp.p_ctg.fa -o ./QC_after_as_raw/BUSCO/BUSCO_HiFiasm_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'\n",
    "\n",
    "input_file = 'assembly.fasta'\n",
    "output_files = 'BUSCO_LJA_eu'\n",
    "command = 'busco -m genome -i ./assembly_new/LJA/assembly.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_LJA_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'\n",
    "\n",
    "input_file = 'Canu_200.contigs.fasta'\n",
    "output_files = 'BUSCO_Canu_200_eu'\n",
    "command ='busco -m genome -i ./assembly_new/Canu_200/Canu_200.contigs.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_Canu_200_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'\n",
    "\n",
    "input_file = 'hifi_wtpoa_cns_200.ctg.fa'\n",
    "output_files = 'BUSCO_Wtdbg_200_eu'\n",
    "command = 'busco -m genome -i ./assembly_new/Wtdbg2_200/cns_200/hifi_wtpoa_cns_200.ctg.fa -o ./QC_after_as_raw/BUSCO/BUSCO_Wtdbg_200_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'\n",
    "\n",
    "input_file = 'assembly.fasta'\n",
    "output_files = 'BUSCO_Flye_eu'\n",
    "command = 'busco -m genome -i ./assembly_new/Flye/assembly.fasta -o ./QC_after_as_raw/BUSCO/BUSCO_Flye_eu -l /media/eternus1/nfs/projects/shared/databases/busco/lineages/eukaryota_odb10'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5224df0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#7. Ragtag\n",
    "\n",
    "#7.1. Download the database\n",
    "\n",
    "output_file ='GCA_018361395.1.zip'\n",
    "command = 'curl -OJX GET \"https://api.ncbi.nlm.nih.gov/datasets/v1/genome/accession/GCA_018361395.1/download?include_annotation_type=GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA&filename=GCA_018361395.1.zip\" -H \"Accept: application/zip\"'\n",
    "\n",
    "#7.2. Unzip the file\n",
    "\n",
    "input_file ='GCA_018361395.1.zip'\n",
    "output_file ='GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna'\n",
    "command = 'unzip GCA_018361395.1.zip'\n",
    "\n",
    "#7.3. Ragtag\n",
    "\n",
    "ragtag.py correct ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna ./assembly_new/Canu/canu_hifi.contigs.fasta -o ./ragtag/Canu_correct -u\n",
    "ragtag.py correct ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna ./assembly_new/HiFiasm/hifi.asm.bp.p_ctg.fa -o ./ragtag/HiFiasm_correct -u\n",
    "ragtag.py correct ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna ./assembly_new/Flye/assembly.fasta -o ./ragtag/Flye_correct -u\n",
    "ragtag.py correct ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna ./assembly_new/Canu_200/Canu_200.contigs.fasta -o ./ragtag/Canu_200_correct -u\n",
    "\n",
    "ragtag.py scaffold ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna ./ragtag/Canu_correct/ragtag.correct.fasta -o ./ragtag/Canu_scaffold -u\n",
    "ragtag.py scaffold ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna ./ragtag/HiFiasm_correct/ragtag.correct.fasta -o ./ragtag/HiFiasm_scaffold -u\n",
    "ragtag.py scaffold ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna  ./ragtag/Flye_correct/ragtag.correct.fasta -o ./ragtag/Flye_scaffold -u\n",
    "ragtag.py scaffold ./curl/ncbi_dataset/data/GCA_018361395.1/GCA_018361395.1_NTU_Bstr_SAD12_2.2_genomic.fna ./ragtag/Canu_200_correct/ragtag.correct.fasta -o ./ragtag/Canu_200_scaffold -u"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7a362641",
   "metadata": {},
   "outputs": [],
   "source": [
    "#8. QUAST after ragtag\n",
    "\n",
    "input_file = 'ragtag.scaffold.fasta'\n",
    "output_files = 'QUAST_Canu_rt'\n",
    "command = 'quast.py -o ./QC_after_rt/QUAST_rt/QUAST_Canu_rt ./ragtag/Canu_scaffold/ragtag.scaffold.fasta'\n",
    "\n",
    "input_file = 'ragtag.scaffold.fasta'\n",
    "output_files = 'QUAST_HiFiasm_rt'\n",
    "command = 'quast.py -o ./QC_after_rt/QUAST_rt/QUAST_HiFiasm_rt ./ragtag/HiFiasm_scaffold/ragtag.scaffold.fasta'\n",
    "\n",
    "input_file = 'ragtag.scaffold.fasta'\n",
    "output_files = 'QUAST_Flye_rt'\n",
    "command = 'quast.py -o ./QC_after_rt/QUAST_rt/QUAST_Flye_rt ./ragtag/Flye_scaffold/ragtag.scaffold.fasta'\n",
    "\n",
    "input_file = 'ragtag.scaffold.fasta'\n",
    "output_files = 'QUAST_Canu_200_rt'\n",
    "command = 'quast.py -o ./QC_after_rt/QUAST_rt/QUAST_Canu_200_rt ./ragtag/Canu_200_scaffold/ragtag.scaffold.fasta'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c776f0e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#9. Phylogeny\n",
    "\n",
    "#9.1. BLAST\n",
    "\n",
    "#9.1.1. Make BLAST database\n",
    "\n",
    "makeblastdb -in ./p_full/GCF_000001735.4_TAIR10.1_protein.faa -parse_seqids -title \"db_TAIR\" -dbtype prot -out ./db_TAIR\n",
    "makeblastdb -in ./p_full/GCF_000004255.2_v.1.0_protein.faa -parse_seqids -title \"db_A_Lyr\" -dbtype prot -out ./BLAST_A_Lyr/db_A_Lyr\n",
    "makeblastdb -in ./p_full/GCF_000309985.2_CAAS_Brap_v3.01_protein.faa -parse_seqids -title \"db_Brap\" -dbtype prot -out ./BLAST_Brap/db_Brap\n",
    "makeblastdb -in ./p_full/GCF_000375325.1_Caprub1_0_protein.faa -parse_seqids -title \"db_Caprub\" -dbtype prot -out ./BLAST_Caprub/db_Caprub\n",
    "makeblastdb -in ./p_full/GCF_000478725.1_Eutsalg1_0_protein.faa -parse_seqids -title \"db_Eutsalg\" -dbtype prot -out ./BLAST_Eutsalg/db_Eutsalg\n",
    "\n",
    "\n",
    "#9.1.2. Search for protein\n",
    "\n",
    "blastp -db ./BLAST_TAIR/db_TAIR -query ./p_apollo/KF705569_1.fasta -out ./BLAST_TAIR/TAIR_coinc\n",
    "blastp -db ./BLAST_A_Lyr/db_A_Lyr -query ./p_apollo/KF705569_1.fasta -out ./BLAST_A_Lyr/Alyr_coinc\n",
    "blastp -db ./BLAST_Brap/db_Brap -query ./p_apollo/KF705569_1.fasta -out ./BLAST_Brap/Brap_coinc\n",
    "blastp -db ./BLAST_Caprub/db_Caprub -query ./p_apollo/KF705569_1.fasta -out ./BLAST_Caprub/Caprub_coinc\n",
    "blastp -db ./BLAST_Eutsalg/db_Eutsalg -query ./p_apollo/KF705569_1.fasta -out ./BLAST_Eutsalg/Eutsalg_coinc\n",
    "\n",
    "#9.2. MUSCLE\n",
    "\n",
    "#9.2.1. Make a FASTA file with all protein sequences\n",
    "\n",
    "input_file = 'all_prot.fasta'\n",
    "output_files = 'align.afa'\n",
    "command = 'muscle -align ./all_prot.fasta -output ./muscle/align.afa'\n",
    "\n",
    "#9.2.2. Error test\n",
    "\n",
    "muscle -align all_prot.fasta -stratified -output ./muscle/error_test/ensemble.efa\n",
    "muscle -disperse ensemble.efa\n",
    "result:  D_LP=0.0005385 D_Cols=0.004032\n",
    "    \n",
    "#9.3. Convertion to Phylip-compatible format\n",
    "\n",
    "#muscle v3!\n",
    "\n",
    "muscle -in align.afa -out align.phylip -phyi\n",
    "    \n",
    "#9.4. Phyml\n",
    "\n",
    "phyml -i align.phylip -d aa -o tlr"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
